tensorboard/plugins/core/core_plugin.py

```diff
@@ -142,27 +142,50 @@ def _serve_runs(self, request):
       A werkzeug Response with the following content:
       {runName: {firstEventTimestamp: 123456.789}}
     """
-    # TODO(chihuahua): When running in database mode, query the Runs table.
-    run_names = sorted(self._multiplexer.Runs())  # Why `sorted`? See below.
-    def get_first_event_timestamp(run_name):
-      try:
-        return self._multiplexer.FirstEventTimestamp(run_name)
-      except ValueError:
-        tf.logging.warning('Unable to get first event timestamp for run %s',
-                           run_name)
-        # Put runs without a timestamp at the end. Their internal
-        # ordering would be nondeterministic, but Python's sorts are
-        # stable, so `sorted`ing the initial list above provides a
-        # deterministic ordering. Of course, we cannot guarantee that
-        # this will be append-only for new event-less runs.
-        return float('inf')
-    first_event_timestamps = {
-        run_name: get_first_event_timestamp(run_name)
-        for run_name in run_names
-    }
-    run_names.sort(key=first_event_timestamps.get)
+    if self._db_connection_provider:
+      db = self._db_connection_provider()
+      cursor = db.execute('''
+        SELECT
+          run_name,
+          IFNULL(started_time, 1e1000) AS started_time
```

Comment:
Maybe add a comment explaining that 1e1000 is a standin for +Inf?  You can do an inline C-style comment e.g.

```sql
/* Use 1e1000 to sort like +Inf (for which there's no literal). */
IFNULL(started_time, 1e1000) AS started_time
```

Also FWIW at least in SQL